src/Evolution/Systems/CurvedScalarWave/Characteristics.hpp

```diff
@@ -0,0 +1,238 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Constraints.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace CurvedScalarWave {
+// @{
+/*!
+ * \ingroup CurvedScalarWave
+ * \brief Compute the characteristic speeds for the scalar wave system in curved
+ * spacetime.
+ *
+ * Computes the speeds as described in "Optimal constraint projection for
+ * hyperbolic evolution systems" by Holst et. al \cite Holst2004wt
+ * [see text following Eq. (32)]. The characteristic fields' names used here
+ * are similar to the paper:
+ *
+ * \f{align*}
+ * \mathrm{SpECTRE} && \mathrm{Holst} \\
+ * v^{\hat \psi} && Z^1 \\
+ * v^{\hat 0}_{i} && Z^{2}_{i} \\
+ * v^{\hat \pm} && u^{1\pm}
+ * \f}
+ *
+ * The corresponding characteristic speeds \f$s\f$ are given in the text
```

Comment:
not a full review yet, but the notation needs to be brought in sync with the flat scalar wave. Here `\lambda` for the speeds